shardingsphere-features/shardingsphere-db-discovery/shardingsphere-db-discovery-provider/shardingsphere-db-discovery-mgr/src/main/java/org/apache/shardingsphere/dbdiscovery/mgr/MGRDatabaseDiscoveryType.java

```diff
@@ -114,6 +115,31 @@ private void checkIsSinglePrimaryMode(final Statement statement) throws SQLExcep
         }
     }
     
+    private void checkDataSourceInReplicationGroup(final Statement statement, final Map<String, DataSource> dataSourceMap) throws SQLException {
+        List<String> memberDataSourceURLs = new LinkedList<>();
```

Comment:
`Collection<String> memberDataSourceURLs` can be `List<String> memberDataSourceURLs`